src/components/generic/movieModal/selectQuality.tsx

```diff
@@ -0,0 +1,82 @@
+import * as React from 'react'
+import { FC, useState, ChangeEvent } from 'react'
+import { Icon } from 'react-icons-kit'
+import { ic_subtitles } from 'react-icons-kit/md/ic_subtitles'
+import { downloadCloud } from 'react-icons-kit/feather/downloadCloud'
+
+// components
+import * as Modal from './style'
+
+interface Link { 
+file: string;
+size_bytes: number;
+torrent_magnet: string;
+}
+
+const SelectQuality:FC<{imdb:string, links:Link[]}> = ({ links, imdb }) => {
+  const [activeItem, setActiveItem] = useState(0)
+
+
+  const toggleLink = (e: ChangeEvent<HTMLSelectElement>) => {
+    let selectValue:number = parseInt(e.target.value, 10)
+    setActiveItem(selectValue)
+  }
+
+  const href =
+    (typeof links[activeItem] !== 'undefined' &&
```

Comment:
I think that `links[activeItem] && links[activeItem].torrent_magnet` will work here, too.

Reply:
Changed it to :
```js
const activeLink = links[activeItem]
  const href = activeLink ? activeLink.torrent_magnet : ""
```